clientv3/concurrency/mutex.go

```diff
@@ -37,35 +41,43 @@ func NewMutex(s *Session, pfx string) *Mutex {
 	return &Mutex{s, pfx + "/", "", -1, nil}
 }
 
+// TryLock locks the mutex if not already locked by another session.
+// If lock is held by another session, return immediately after attempting necessary cleanup
+func (m *Mutex) TryLock(ctx context.Context) error {
```

Comment:
TryLock is supposed to be non-blocking. So passing in a ctx is kind of confusing. We need to comment that the ctx only controls the network timeout not the lock waiting timeout (also verify if this is true?).

Reply:
Yeah let's document the `ctx` argument. It's only used for sending/receiving Txn RPC.

Reply:
Agree that `ctx` is only for timeout on RPC and not for waiting for lock to be acquired.

Reply:
If we could change the semantics a little, we can have only one `Mutex.Lock` which returns two values `(err error,   waitFunc func(context) error)`. The `err` can signal if locking was successful or not, and if any context timeout occured. The user now has choice to wait for lock by calling the `waitFunc` with a new context, same behavior as `Mutex.Lock`. If user decides not to invoke `waitFunc` it is like `Mutext.TryLock`. This way the purpose of `ctx` is also clear, and there is less code.
Let me know your thoughts. It would surely be a breaking change.

Reply:
Not sure if my idea is better or worse: keep the original Lock function and its semantics, but have a new function which returns (err, waitFunc).

Reply:
i will take this up in a separate PR